File: SiriusJavaClient-Win32/src/main/java/org/sirius/client/win32/classes/TopLevelWindow.java
/**
 * .
 */
package org.sirius.client.win32.classes;

import org.sirius.client.win32.Win32Client;
import org.sirius.client.win32.types.Win32Locator;

/**
 * @author Myk Kolisnyk
 * .
 */
public class TopLevelWindow extends MovableWindow {

    /**
     * @param clientValue .
     * @param locatorValue .
     */
    public TopLevelWindow(
            final Win32Client clientValue,
            final Win32Locator locatorValue) {
        super(clientValue, locatorValue);
    }

    /**
     * .
     * @param clientValue .
     * @param parentValue .
     * @param locatorValue .
     */
    public TopLevelWindow(
            final Win32Client clientValue,
            final Window parentValue,
            final Win32Locator locatorValue) {
        super(clientValue, parentValue, locatorValue);
    }

    /**
     * .
     * @return .
     * @throws Exception .
     */
    public final Menu menu() throws Exception {
        if (this.exists()) {
            Menu menu = new Menu(client, this);
            return menu;
        }
        return null;
    }

    /**
     * .
     * @throws Exception .
     */
    public final void setActive() throws Exception {
        client.core().window().activate(locator.getHwnd());
    }

    /**
     * .
     * @return .
     * @throws Exception .
     */
    public final Menu systemMenu() throws Exception {
        if (this.exists()) {
            long hmenu = client.core().window()
                    .getSystemMenu(this.getHwnd(), true);
            Menu menu = new Menu(client, this, hmenu);
            return menu;
        }
        return null;
    }
}

File: sirius-ruby-client-win32/lib/sirius/win32/classes/TopLevelWindow.rb
module Sirius
  module Client
    module Win32
      module Classes
        class TopLevelWindow < MovableWindow
          def initialize(client=nil, locator=nil,parent=nil)
            super(client, parent, locator)
          end

          def set_active
            @client.core.window.activate(@locator.hwnd)
          end

          def menu
            if exists?
              return Menu.new(client,self)
            end
            return nil
          end

          def system_menu
            if (exists?)
              @hmenu = @client.core.window.getSystemMenu(@locator.hwnd, true);
              return Menu.new(client, self, @hmenu)
            end
            return nil
          end
        end
      end
    end
  end
end